### PR TITLE
add mcm foudnation prow job

### PR DESCRIPTION
**Pull request by qiujian16**
**What this PR does / why we need it**:
Add mcm foundation prow job 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Specify your PR reviewers**:

/cc @gyliu513 @morvencao

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

File: prow/cluster/jobs/rh-ibm-synergy/multicloud-operators-foundation/IBM.multicloud-operators-foundation.master.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            secretKeyRef:
+              name: ibm-multicloud-operators-foundation-codecov-token
+              key: codecov-token
+        securityContext:
+          privileged: true
+  - name: build_multicloud-operators-foundation_postsubmit
+    branches:
+    - ^master$
+    decorate: true
+    path_alias: github.com/IBM/multicloud-operators-foundation
```
Comment (morvencao) [inline prow/cluster/jobs/rh-ibm-synergy/multicloud-operators-foundation/IBM.multicloud-operators-foundation.master.yaml]: s/IBM/rh-ibm-synergy

File: prow/cluster/jobs/rh-ibm-synergy/multicloud-operators-foundation/IBM.multicloud-operators-foundation.master.yaml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        name: ""
+        securityContext:
+          privileged: true
+    trigger: '(?m)^/test (?:.*? )?build(?: .*?)?$'
+  - name: test_multicloud-operators-foundation
+    branches:
+    - ^master$
+    decorate: true
+    always_run: true
+    path_alias: github.ibm.com/IBMPrivateCloud/multicloud-operators-foundation
```
Comment (morvencao) [inline prow/cluster/jobs/rh-ibm-synergy/multicloud-operators-foundation/IBM.multicloud-operators-foundation.master.yaml]: See comment about path_alias above.

File: prow/cluster/jobs/rh-ibm-synergy/multicloud-operators-foundation/IBM.multicloud-operators-foundation.master.yaml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+          privileged: true
+    trigger: '(?m)^/test (?:.*? )?test(?: .*?)?$'
+
+postsubmits:
+  IBM/multicloud-operators-foundation:
+  - name: test_multicloud-operators-foundation_postsubmit
+    branches:
+    - ^master$
+    decorate: true
+    path_alias: github.ibm.com/IBMPrivateCloud/multicloud-operators-foundation
```
Comment (morvencao) [inline prow/cluster/jobs/rh-ibm-synergy/multicloud-operators-foundation/IBM.multicloud-operators-foundation.master.yaml]: Same issue for path_alias.

File: prow/cluster/jobs/rh-ibm-synergy/multicloud-operators-foundation/IBM.multicloud-operators-foundation.master.yaml [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        - test
+        image: quay.io/multicloudlab/build-tools:2019-10-08T07-03-25
+        name: ""
+        securityContext:
+          privileged: true
+  - name: coverage_multicloud-operators-foundation_postsubmit
+    branches:
+    - ^master$
+    decorate: true
+    path_alias: github.ibm.com/IBMPrivateCloud/multicloud-operators-foundation
```
Comment (morvencao) [inline prow/cluster/jobs/rh-ibm-synergy/multicloud-operators-foundation/IBM.multicloud-operators-foundation.master.yaml]: Same issue for path_alias.

File: prow/cluster/jobs/rh-ibm-synergy/multicloud-operators-foundation/IBM.multicloud-operators-foundation.master.yaml [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+        - make
+        - coverage
+        image: quay.io/multicloudlab/build-tools:2019-10-08T07-03-25
+        name: ""
+        env:
+        - name: CODECOV_TOKEN
+          valueFrom:
+            secretKeyRef:
+              name: ibm-multicloud-operators-foundation-codecov-token
+              key: codecov-token
```
Comment (morvencao) [inline prow/cluster/jobs/rh-ibm-synergy/multicloud-operators-foundation/IBM.multicloud-operators-foundation.master.yaml]: To enable CODECOV, you need to authorize https://codecov.io/ to access your repo and generate code coverage upload token, and then I can help you to create CODECOV_TOKEN  secret, ping me in slack channel.

File: prow/cluster/jobs/rh-ibm-synergy/multicloud-operators-foundation/IBM.multicloud-operators-foundation.master.yaml [hunk 6]
```diff
@@ -0,0 +1,87 @@
+presubmits:
+  rh-ibm-synergy/multicloud-operators-foundation:
+  - name: build_multicloud-operators-foundation
+    branches:
+    - ^master$
+    decorate: true
+    always_run: true
+    path_alias: github.ibm.com/IBMPrivateCloud/multicloud-operators-foundation
```
[the remark this answers is not in the document]
Comment (qiujian16) [reply prow/cluster/jobs/rh-ibm-synergy/multicloud-operators-foundation/IBM.multicloud-operators-foundation.master.yaml]: we need `github.ibm.com/IBMPrivateCloud/multicloud-operators-foundation` to be import path...`github.com/rh-ibm-synergy/multicloud-operators-foundation` is just the code repo

File: prow/cluster/jobs/rh-ibm-synergy/multicloud-operators-foundation/IBM.multicloud-operators-foundation.master.yaml (hunk 5 above)
Comment (kuanf) [reply prow/cluster/jobs/rh-ibm-synergy/multicloud-operators-foundation/IBM.multicloud-operators-foundation.master.yaml]: I don't think we want to enable CODECOV at this stage, the codecov report contains file diff and disclosues code.  We shall enable this after it is fully public.

Comment (morvencao) [reply prow/cluster/jobs/rh-ibm-synergy/multicloud-operators-foundation/IBM.multicloud-operators-foundation.master.yaml]: Correct, let's remove this section for now.

Comment (qiujian16) [reply prow/cluster/jobs/rh-ibm-synergy/multicloud-operators-foundation/IBM.multicloud-operators-foundation.master.yaml]: done

File: prow/cluster/jobs/rh-ibm-synergy/multicloud-operators-foundation/IBM.multicloud-operators-foundation.master.yaml [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+        - make
+        - test
+        image: quay.io/multicloudlab/build-tools:2019-10-08T07-03-25
+        name: ""
+        securityContext:
+          privileged: true
+    trigger: '(?m)^/test (?:.*? )?test(?: .*?)?$'
+
+postsubmits:
+  IBM/multicloud-operators-foundation:
```
[the remark this answers is not in the document]
Comment (qiujian16) [reply prow/cluster/jobs/rh-ibm-synergy/multicloud-operators-foundation/IBM.multicloud-operators-foundation.master.yaml]: done

**Pull request merged**